cmd/mdatagen/internal/templates/telemetry.go.tmpl

```diff
@@ -134,9 +135,16 @@ func NewTelemetryBuilder(settings component.TelemetrySettings, options ...Teleme
     builder.meter = Meter(settings)
     var err, errs error
 
+    var name string
     {{- range $name, $metric := .Telemetry.Metrics }}
+    name = "otelcol_{{ $name }}"
+    featuregate.GlobalRegistry().VisitAll(func(gate *featuregate.Gate) {
```

Comment:
I think this is reasonable, an alternative would be to defined the gate in an internal module and then import it here. I am personally fine with this approach, even if awkward there should only be a handful feature gates at any given time.

Reply:
I don't think an internal module works here because the requirement is that the feature gate can be set at run time, so knowledge of the feature gate must be part of the generated code. If it's part of the generated code, then it can't be internal to this repository because then no components outside the repository can import the package.

Reply:
Ah, you are right, we need to expose _something_ so that the generated code can access it. Anyway, I think this solution is fine.